echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/config/SlackConfig.groovy

```diff
@@ -40,38 +45,49 @@ import static retrofit.Endpoints.newFixedEndpoint
 @Slf4j
 @CompileStatic
 class SlackConfig {
-
-  final static String SLACK_INCOMING_WEBHOOK = 'https://hooks.slack.com/services'
+  final static String SLACK_INCOMING_WEBHOOK = 'https://hooks.slack.com'
   final static String SLACK_CHAT_API = 'https://slack.com'
 
-  @Value('${slack.base-url:}')
-  String slackBaseUrl
+  @ConfigurationProperties(prefix = "slack")
+  @Component
+  class SlackProperties {
+    private String _baseUrl
```

Comment:
why the `_` here?

Reply:
Because the actual property name is `baseUrl` which is now accessible via the getter/setter. This would much easier if it was Kotlin, but it's Groovy, so I had to use this trick... This is an example of refactoring how the Slack config was implemented but keeping the existing contract.

Reply:
I don't follow. Spring (should) see that there's a setter and use that, rather than using reflection to set the property directly.

Reply:
The catch is that I need to be able to set the property but manipulate the return in the getter. I did this to maintain backwards compatibility with how config was being used.